modules/seastate/src/SeaSt_WaveField.f90

```diff
@@ -432,6 +423,442 @@ SUBROUTINE WaveField_GetWaveKin( WaveField, SeaSt_Interp_m, Time, pos, forceNode
       END IF
    END DO
 
-END SUBROUTINE WaveField_GetWaveKin
+contains
+   logical function Failed()
+      call SetErrStat( ErrStat2, ErrMsg2, ErrStat, ErrMsg, RoutineName )
+      Failed = ErrStat >= AbortErrLev
+   end function
+end subroutine WaveField_GetWaveKin
+
+
+!----------------------------------------------------------------------------------------------------
+! Interpolation related functions
+!----------------------------------------------------------------------------------------------------
+
+!> Set the WaveField 4D Params
+subroutine WaveField_SetParam( InitInp, p )
```

Comment:
I'd be inclined to remove the `InitInp` type from the WaveField module. We set a lot of the WaveField parameters in different routines, and this one only sets values that were in the SeaState_Interp module. WaveField also didn't have any InitInp types before this.

Reply:
Thanks for looking at this PR!

I agree; `InitInp` really doesn't need to exist at all.  I debated removing this at the time, but in the end copied over the `InitInput` from the `SeaState_Interp` for simplicity.

While we could remove the `SetParam` routine as well, I'm inclined to leave the it to force future programmers to set all the parameters.

Reply:
I'm not sure we can set all the parameters for WaveField in this routine without rewriting all of SeaState from scratch. :-) Most of the SeaState, Waves, and Waves2 init routine set different parameters in WaveField. It gets particularly tricky in the Waves modules where parameters get set differently based on which type of waves are being modeled. If you're going to keep this routine, maybe rename it?

Reply:
Removed!